### ISSUE TITLE
Select: Safari VoiceOver does not announce selected value

### ISSUE THREAD

**Issue by jramke** [chakra-ui/zag#3099]
# 🐛 Bug report

When using the Select machine (e.g docs example) the selected value is not correctly announced when using VoiceOver in Safari Tahoe (didnt test other versions). 

When selecting a value and the focus moves back to the trigger the default value "select option" is announced. 

During another testing i also noticed that not the default but the previous option was announced when selecting a new value. However i cant reproduce this. For the first case you can see a screen recording below.

## 💥 Steps to reproduce

Open the docs select example and select a value using VO.

## 💻 Link to reproduction

https://zagjs.com/components/select

## 🧐 Expected behavior

The updated value/content of the trigger should be announced.

## 🧭 Possible Solution

In my case i made the open state controlled and used a timeout like this. I dont know why the raf in the machine is not enough.
```ts
onOpenChange(details) {
	setTimeout(() => {
		setOpen(details.open);
	}, 0);
},
```

i also noticed that safari VO sometimes read the first value of the content list (although display: hidden) because of the aria-controls attiribute linking to it. This could be resolved when i removed the aria-controls attribute when the popover is not open, maybe this could also help to fix some edgecases and it would not break anything.
```ts
const triggerProps = api.getTriggerProps();
const mergedTriggerProps = mergeProps(triggerProps, {
	...(!api.open ? { 'aria-controls': null } : {}),
});
```

## 🌍 System information

| Software         | Version(s) |
| ---------------- | ---------- |
| Zag Version      |       1.40.0     |
| Browser          |       Safari 26.3     |
| Operating System |      Tahoe 26.3      |

## 📝 Additional information

https://github.com/user-attachments/assets/09dbac90-6e22-4db5-8ddf-95ae934049e1

**Comment by segunadebayo** [comment]
This issue has been automatically closed due to inactivity. If you believe this closure was in error, please reopen the issue or add a comment. Thanks for your understanding.

Note: This is an automated message.

**Comment by jramke** [comment]
I had time poking around a little in the machine source and noticed that also replacing the raf in the focusTriggerEl action with a setTimeout or nextTick is not sufficient for safari.
It seems safari simply ignores/caches the span elements value that contains the value text. When using an aria-label on the button or no span and setting the buttons text value directly, it works. Maybe this helps anyone coming across this weird safari gotcha in future

**Comment by jramke** [comment]
or simply forcing the node to rerender works too `<span key={api.valueAsString}>`.

I think its worth mentioning this or making it default in examples and include it in ark.
